tests/ptvsd/server/test_flask.py

```diff
@@ -29,56 +29,54 @@ class lines:
     app_py = code.get_marked_line_numbers(paths.app_py)
 
 
-def _initialize_session(session, multiprocess=None, exit_code=0):
-    if multiprocess:
-        pytest.skip("https://github.com/microsoft/ptvsd/issues/1706")
-
-    env = {
-        "FLASK_APP": paths.app_py,
-        "FLASK_ENV": "development",
-        "FLASK_DEBUG": "1" if multiprocess else "0",
-    }
-    if platform.system() != "Windows":
-        locale = "en_US.utf8" if platform.system() == "Linux" else "en_US.UTF-8"
-        env.update({"LC_ALL": locale, "LANG": locale})
-
-    args = ["run"]
-    if not multiprocess:
-        args += ["--no-debugger", "--no-reload", "--with-threads"]
-    args += ["--port", str(flask.port)]
-
-    session.expected_exit_code = exit_code
-    session.configure(
-        "module",
-        "flask",
-        cwd=paths.flask1,
-        jinja=True,
-        subProcess=multiprocess,
-        args=args,
-        env=env,
-    )
-
-
-@pytest.mark.parametrize(
-    "start_method", [runners.launch, runners.attach_by_socket["cli"]]
-)
+@pytest.fixture
+@pytest.mark.parametrize("run", [runners.launch, runners.attach_by_socket["cli"]])
+def start_flask(run):
+    def start(session, multiprocess=False):
+        if multiprocess:
+            pytest.skip("https://github.com/microsoft/ptvsd/issues/1706")
+
+        env = {
```

Comment:
It doesn't make a difference in this case, but it's generally better to access `session.config.env` directly rather than setting it via `config["env"]`, because you get some extra features (like PYTHONPATH handling) there. So it might be best to do it in all tests for consistency, as they get copy/pasted around a lot when writing new ones.

Reply:
(and can you also do this to `test_django`?)

Reply:
We don't really use env in django. Only flask requires FLASK_APP to be set via env.

Reply:
I think we should entirely block people from using `config["env"]`. Essentially using that should fail with `Use config.env instead.` witha comment explaining why.